client/sites-dashboard/components/sites-dashboard.tsx

```diff
@@ -210,6 +216,17 @@ export function SitesDashboard( {
 						>
 							<span>{ __( 'Import an existing site' ) }</span>
 						</PopoverMenuItem>
+						<PopoverMenuSeparator />
+						<PopoverMenuItem icon="search" href={ domainAddNew() } onClick={ trackAddDomainClick }>
+							<span>{ __( 'Search for a domain' ) }</span>
+						</PopoverMenuItem>
+						<PopoverMenuItem
+							icon="domains"
+							onClick={ () => trackAddDomainMenuClick( domainUseMyDomain() ) }
+							href={ domainUseMyDomain() }
```

Comment:
In my local environment, this doesn't appear to behave as expected.

I'm redirected to `/domains/add/use-my-domain/undefined`, at which point I'm then redirected to `/domains/add`